dht/token-sharding.hh

```diff
@@ -87,4 +87,20 @@ inline std::ostream& operator<<(std::ostream& os, const sharder& sharder) {
     return os;
 }
 
+/*
+ * Finds the first token in token range (`start`, `end`] that belongs to shard shard_idx.
+ *
+ * If there is no token that belongs to shard shard_idx in this range,
+ * `end` is returned.
```

Comment:
How does one distingish between `end` used as a failure marker and `end` being a valid result?

Better return std::optional instead.

Reply:
done.